subsys/bluetooth/host/gatt.c

```diff
@@ -5547,6 +5717,15 @@ void bt_gatt_encrypt_change(struct bt_conn *conn)
 #endif	/* CONFIG_BT_GATT_AUTO_RESUBSCRIBE */
 
 	bt_gatt_foreach_attr(0x0001, 0xffff, update_ccc, &data);
+
+#if defined(CONFIG_BT_SETTINGS)
+	if (!bt_gatt_change_aware(conn, false)) {
+		/* Send a Service Changed indication if the current peer is
+		 * marked as change-unaware.
+		 */
+		sc_indicate(0x0001, 0xffff);
```

Comment:
We could consider optimizing it and indicating only to this particular peer (may be done in the future since it should not lead to bugs - it's just an optimization)

Reply:
yeah I wanted to do that originally, but its trickier than it seems because we use global flags, not per-peer for book-keeping. Maybe will do the optimization later.

Reply:
yep, will do that in the future. Please open an Enhancement issue linking this comment, you can assign it to me :+1:

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/54766